dot/peerset/peerstate.go

```diff
@@ -237,17 +238,17 @@ func (ps *PeersState) hasFreeIncomingSlot(set int) bool {
 
 // addNoSlotNode adds a node to the list of nodes that don't occupy slots.
 // has no effect if the node was already in the group.
-func (ps *PeersState) addNoSlotNode(idx int, peerID peer.ID) {
+func (ps *PeersState) addNoSlotNode(idx int, peerID peer.ID) error {
```

Comment:
could we add a condition `if idx >= len(ps.sets)` then return an error? it could not happen but just to adopt a kind of defensive programming.